src/commonMain/kotlin/sim/rotation/criteria/ResourcePctGte.kt

```diff
@@ -15,7 +16,25 @@ class ResourcePctGte(data: RotationRuleCriterion) : Criterion(Type.RESOURCE_PCT_
         null
     }
 
+    inline fun <reified T : Enum<T>> valueOf(type: String): T? {
+        return java.lang.Enum.valueOf(T::class.java, type)
```

Comment:
We cannot use Java types, as it would be non-portable to the JS build.  A companion method on `Resource` named `getResourceFromString(str: String): Resource?` or something like that would probably be best.